Feature - Views and navigation

wickathou opened the pull request:
# Features
- [x] All views for Posts and Users are linked
- [x] Views render dynamic data from the database

wickathou committed:
Added rails-controller-testing to test template rendering
Modified Post and User controller to return json or html depending of the request
Created tests for Post and User route requests

wickathou committed:
modified readme to include the terminal command to run the unit tests

wickathou committed:
Added missing cursor pointer for link_to elements

t-yanick commented on app/views/users/show.html.erb:
```diff
@@ -1,5 +1,5 @@
 <%= render partial: "partials/user_listing", locals: {user: @user}%>
 <%= render partial: "partials/user_details"%>
```
- [x] From your user_details partial, you are using <% @posts.each do |post| %> but for this show page, you should be displaying information or details of recents posts. You should therefore make use of your `def last_three_posts `method you created in the user model

```
# app/views/users/show.html.erb

[...]

<% @user.recent_post.each do |post| %>

<h4><%= link_to post.title, user_post_path(post) %></h4>
<p><%= post.text %></p>
<p>
<small>Comments: <%= post.comments_counter || 0 %>,</small> 
<small>Likes: <%= post.likes_counter || 0 %></small>
</p>

<% end %>

<button>
<%= link_to 'See all Post', user_posts_path(@user) %>
</button>

```

rloterh replied:
well resolved! ✔️

t-yanick commented on app/views/posts/index.html.erb:
```diff
@@ -6,5 +6,5 @@
   <% end %>
 </div>
```
- [x] In the _comments partial, you should display both the author name and text. At the moment, your code is displaying just the text
`<p><%= comment.author.name %>: <%= comment.text %></p>`

rloterh replied:
well resolved! ✔️

t-yanick reviewed (changes requested):
# STATUS: CHANGES REQUIRED

Hi @wickathou , 

### Good job so far!
### To Highlight! :clap:  :green_circle: 

:heavy_check_mark: Excellent use of partials
:heavy_check_mark: Use of correct gitflow
:heavy_check_mark:  Linters all passing
:heavy_check_mark:  Good commit messages
:heavy_check_mark: Descriptive Readme file

There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️

- [x] In your _user_listing partial, your code is missing the display of the user photo. Kindly review that.

_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

wickathou committed:
Added user name to comments rendered
Modified user details to preview only the last three posts
Modified user listing to preview user picture

rloterh reviewed (approved):
## Status: Approved :heavy_check_mark:

Hi Javier,  

Your project is complete! 🏁 There is nothing else to say other than... it's time to merge it 🍾 Congratulations!  🎉💯🌟 

## Highlights 🎯 
🎲 Good job for introducing improvements to the views!
🎲 Good ReadMe!

Cheers..Happy coding! 💻 🍷 🚀 

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
